homeassistant/components/systemmonitor/sensor.py

```diff
@@ -70,185 +77,367 @@
+    disk = cast(sdiskusage, entity.coordinator.data)
+    return round(disk.used / 1024**3, 1)
+
+
+def get_disk_free(entity: SystemMonitorSensor[dataT]) -> float:
+    """Return disk free."""
+    disk = cast(sdiskusage, entity.coordinator.data)
+    return round(disk.free / 1024**3, 1)
+
+
+def get_memory_use_percent(entity: SystemMonitorSensor[dataT]) -> float:
+    """Return memory use percent."""
+    virtual_memory = cast(svmem, entity.coordinator.data)
+    return virtual_memory.percent
+
+
+def get_memory_use(entity: SystemMonitorSensor[dataT]) -> float:
+    """Return memory use."""
+    virtual_memory = cast(svmem, entity.coordinator.data)
+    return round((virtual_memory.total - virtual_memory.available) / 1024**2, 1)
+
+
+def get_memory_free(entity: SystemMonitorSensor[dataT]) -> float:
+    """Return memory free."""
+    memory = cast(svmem, entity.coordinator.data)
+    return round(memory.available / 1024**2, 1)
+
+
+def get_swap_use_percent(entity: SystemMonitorSensor[dataT]) -> float:
+    """Return swap use percent."""
+    swap = cast(sswap, entity.coordinator.data)
+    return swap.percent
+
+
+def get_swap_use(entity: SystemMonitorSensor[dataT]) -> float:
+    """Return swap use."""
+    swap = cast(sswap, entity.coordinator.data)
+    return round(swap.used / 1024**2, 1)
+
+
+def get_swap_free(entity: SystemMonitorSensor[dataT]) -> float:
+    """Return swap free."""
+    swap = cast(sswap, entity.coordinator.data)
+    return round(swap.free / 1024**2, 1)
+
+
+def get_processor_use(entity: SystemMonitorSensor[dataT]) -> int:
+    """Return processor use."""
+    return round(cast(float, entity.coordinator.data))
+
+
+def get_processor_temperature(entity: SystemMonitorSensor[dataT]) -> float | None:
+    """Return processor temperature."""
+    temps = cast(dict[str, list[shwtemp]], entity.coordinator.data)
+    temp = read_cpu_temperature(temps)
+    return temp
+
+
+def get_process(entity: SystemMonitorSensor[dataT]) -> str:
+    """Return process."""
+    state = STATE_OFF
+    for proc in cast(Iterator[psutil.Process], entity.coordinator.data):
+        _LOGGER.debug("process %s for argument %s", proc.name(), entity.argument)
+        try:
+            if entity.argument == proc.name():
+                state = STATE_ON
+                break
+        except psutil.NoSuchProcess as err:
+            _LOGGER.warning(
+                "Failed to load process with ID: %s, old name: %s",
+                err.pid,
+                err.name,
+            )
+    return state
+
+
+def get_network(entity: SystemMonitorSensor[dataT]) -> float | None:
+    """Return network in and out."""
+    counters = cast(dict[str, snetio], entity.coordinator.data)
+    if entity.argument in counters:
+        counter = counters[entity.argument][IO_COUNTER[entity.entity_description.key]]
+        return round(counter / 1024**2, 1)
+    return None
+
+
+def get_packets(entity: SystemMonitorSensor[dataT]) -> float | None:
```

Comment:
Why is this function just returning the result of `get_network`? Ie, why do we need it?

Reply:
We don't. I just created them one by one as a first step to make sure it all works before cleaning it up and consolidating where possible.
Need to solve some issues first, see https://github.com/home-assistant/core/pull/106693#discussion_r1446666406